doc/memos/rdm-draft-alamos-ieee802154-radio-hal.md

```diff
@@ -0,0 +1,967 @@
+     *        frame pending bit, status, etc).
+     *
+     * @pre the device is on
+     *
+     * @note this function pointer can be NULL if the device doesn't support
+     *       frame retransmissions
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[out] info the TX information
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*get_tx_status)(ieee802154_dev_t *dev, ieee802154_tx_info_t *info);
+
+    /**
+     * @brief Set IEEE802.15.4 address in hardware address filter
+     *
+     * @pre the device is on
+     *
+     * @note this function pointer can be NULL if the device doesn't support
+     *       hardware address filtering.
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] short_addr the IEEE802.15.4 short address
+     * @param[in] ext_addr the IEEE802.15.4 extended address
+     * @param[in] pan_id the IEEE802.15.4 PAN ID
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*set_hw_addr_filter)(ieee802154_dev_t *dev, uint8_t *short_addr,
+                              uint8_t *ext_addr, uint16_t pan_id);
+
+    /**
+     * @brief Set number of frame retransmissions
+     *
+     * @pre the device is on
+     *
+     * @note this function pointer can be NULL if the device doesn't support
+     *       frame retransmissions
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] retries the number of retransmissions
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*set_frame_retries)(ieee802154_dev_t *dev, int retries);
+
+    /**
+     * @brief Set IEEE802.15.4 CSMA configuration parameters
+     *
+     * @pre the device is on
+     *
+     * @note this function pointer can be NULL if the device doesn't support
+     *       frame retransmissions
+     *
+     * @param[in] dev IEEE802.15.4 device descriptor
+     * @param[in] bd parameters of the exponential backoff
+     * @param[in] retries number of CSMA-CA retries. If @p retries < 0, 
+     *                    retransmissions with CSMA-CA MUST be disabled.
+     *
+     * @return 0 on success
+     * @return negative errno on error
+     */
+    int (*set_csma_params)(ieee802154_dev_t *dev, ieee802154_csma_be_t *bd,
+                           int8_t retries);
+```
+
+## 6.2 Event Notification
+
+The Event Notification mechanism is implemented with a function callback.
+The callback function is supposed to be implemented by the upper layer.
+
+The callback signature, the events and their expected behavior are defined
+in the following block:
+
+```c
+/**
+ * @brief   Event notification status
+ */
+typedef enum {
+    /**
+     * @brief the transceiver detected a valid Start Frame Delimiter, which
+              indicates the radio started to receive a packet.
+     */
+    IEEE802154_RADIO_RX_START,
```

Comment:
Could we add `IEEE802154_RADIO_TX_START` (and the respective CAP), this information is important for getting accurate timing for 6TSCH or UWB ranging.

Reply:
yes, sure!